src/StripeTests/Services/OAuth/OAuthTokenServiceTest.cs

```diff
@@ -0,0 +1,89 @@
+namespace StripeTests
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+
+    using Stripe;
+    using Xunit;
+
+    public class OAuthTokenServiceTest : BaseStripeTest
+    {
+        private const string OrderId = "or_123";
+
+        private readonly OAuthTokenService service;
+        private readonly OAuthTokenCreateOptions createOptions;
+        private readonly OAuthTokenDeauthorizeOptions deauthorizeOptions;
+
+        public OAuthTokenServiceTest(MockHttpClientFixture mockHttpClientFixture)
+            : base(mockHttpClientFixture)
+        {
+            this.service = new OAuthTokenService();
+
+            this.createOptions = new OAuthTokenCreateOptions
+            {
+                Code = "this_is_an_authorization_code",
+                GrantType = "authorization_code",
+            };
+
+            this.deauthorizeOptions = new OAuthTokenDeauthorizeOptions
+            {
+                StripeUserId = "acct_test_deauth",
+            };
+        }
+
+        [Fact]
+        public void Create()
+        {
+            // stripe=mock doesn't support OAuth endpoints, so stub the request
+            var json = GetResourceAsString("oauth_fixtures.token_response.json");
+            this.StubRequest(HttpMethod.Post, "/oauth/token", HttpStatusCode.OK, json);
+
+            var oauthToken = this.service.Create(this.createOptions);
+            this.AssertRequest(HttpMethod.Post, "/oauth/token");
```

Comment:
which parts double check that we use the right URL path?

Reply:
`AssertRequest` checks the HTTP method and path, but nothing else (i.e. the request parameters). I'll try to add support for it in the future.

(Also, `StubRequest` implicitly checks the method and path as well -- if the request weren't stubbed with the correct method and path, the call to `Create` would send a real request to the API and fail.)